src/site/topic19.rst

```diff
@@ -117,10 +117,78 @@ Factorial
 Recursive Programming
 =====================
 
-
 Factorial
 ---------
 
+* The factorial,  :math:`n!`, of a non-negative integer is the product of all non-negative between n and 1 inclusively
```

Comment:
0! = 1 is not captured here. And fun fact, factorial is defined for negative values and all complex numbers. I imagine simplest here is to say we define factorial for positive integers.

Reply:
Spoke offline about gamma, so let's move on from that. For this, you could also define factorial as `0! = 1, n! = n * (n-1)!`

Reply:
The recursive definition is below. this line is to start with the high-level idea.